src/backend/logger/logger.ts

```diff
@@ -468,6 +474,128 @@ class GameLogWriter extends LogWriter {
+        delete gameSettings['winePrefix']
+        delete gameSettings['autoInstallDxvk']
+        delete gameSettings['autoInstallDxvkNvapi']
+        delete gameSettings['autoInstallVkd3d']
+        delete gameSettings['enableFsync']
+        delete gameSettings['enableEsync']
+        delete gameSettings['enableFSR']
+        delete gameSettings['showFps']
+        delete gameSettings['enableDXVKFpsLimit']
+        delete gameSettings['eacRuntime']
+        delete gameSettings['battlEyeRuntime']
+        delete gameSettings['useGameMode']
+      }
+    }
+
+    // remove settings that are not used on Mac
+    if (isMac) {
+      delete gameSettings['useGameMode']
+      delete gameSettings['gamescope']
+      delete gameSettings['nvidiaPrime']
+      delete gameSettings['battlEyeRuntime']
+      delete gameSettings['eacRuntime']
+      delete gameSettings['enableFSR']
+      delete gameSettings['showMangohud']
+      delete gameSettings['showFps']
+
+      if (notNative) {
+        const wineType = gameSettings['wineVersion']
+        if (wineType) {
+          if (wineType['type'] === 'wine') {
+            delete gameSettings['wineCrossoverBottle']
+          }
+
+          if (wineType['type'] === 'toolkit') {
+            delete gameSettings['autoInstallDxvk']
+            delete gameSettings['autoInstallDxvkNvapi']
+            delete gameSettings['autoInstallVkd3d']
+          }
+
+          if (wineType['type'] === 'crossover') {
+            delete gameSettings['autoInstallDxvk']
+            delete gameSettings['autoInstallDxvkNvapi']
+            delete gameSettings['autoInstallVkd3d']
+          }
+        }
+
+        delete gameSettings['wineVersion']
+        delete gameSettings['winePrefix']
+      } else {
+        // remove settings that are not used on native Mac games
+        delete gameSettings['enableDXVKFpsLimit']
+        delete gameSettings['wineVersion']
+        delete gameSettings['winePrefix']
+        delete gameSettings['wineCrossoverBottle']
+      }
+    }
+
+    // remove settings that are not used on Windows
+    if (isWindows) {
+      delete gameSettings['enableMsync']
+      delete gameSettings['enableFSR']
+      delete gameSettings['enableEsync']
+      delete gameSettings['enableFsync']
+      delete gameSettings['enableDXVKFpsLimit']
+      delete gameSettings['DXVKFpsCap']
+      delete gameSettings['autoInstallDxvk']
+      delete gameSettings['autoInstallDxvkNvapi']
+      delete gameSettings['autoInstallVkd3d']
+      delete gameSettings['gamescope']
+      delete gameSettings['useGameMode']
+      delete gameSettings['showMangohud']
+      delete gameSettings['showFps']
+      delete gameSettings['preferSystemLibs']
+      delete gameSettings['wineCrossoverBottle']
+      delete gameSettings['winePrefix']
+      delete gameSettings['wineVersion']
+      delete gameSettings['battlEyeRuntime']
+      delete gameSettings['eacRuntime']
+      delete gameSettings['nvidiaPrime']
+    }
```

Comment:
This whole function should use dot notation to access properties. Bracket notation currently hides some errors, since we have `noUncheckedIndexedAccess` turned on in our tsconfig